src/endpoints/blocks/block.service.ts

```diff
@@ -58,17 +58,16 @@ export class BlockService {
     const proposerBlses = blocks.map(x => x.proposer);
 
     const nodes = await this.nodeService.getAllNodes();
-    for (const node of nodes) {
-      if (!proposerBlses.includes(node.bls)) {
-        continue;
-      }
+    const relevantNodes = nodes.filter(node => proposerBlses.includes(node.bls) && node.identity);
 
-      const nodeIdentity = node.identity;
-      if (!nodeIdentity) {
-        continue;
-      }
+    const identityPromises = relevantNodes.map(async (node) => {
+      const identity = await this.identitiesService.getIdentity(node.identity!);
+      return { node, identity };
+    });
+
+    const nodeIdentities = await Promise.all(identityPromises);
```

Comment:
we currently have 171 identities on mainnet. I don't think ES or other underlying services will handle this concurrent load that well. we can use a sort of concurrency limiting solution, such as [p-limit](https://doziestar.medium.com/serving-tasks-efficiently-understanding-p-limit-in-javascript-fb524a35b846), Bluebird or something custom.